packager/signer/signer.go

```diff
@@ -128,13 +128,25 @@ func noRedirects(req *http.Request, via []*http.Request) error {
 	return http.ErrUseLastResponse
 }
 
-func New(cert *x509.Certificate, key crypto.PrivateKey, urlSets []util.URLSet,
+func New(cert *x509.Certificate, key crypto.PrivateKey, docroot string, urlSets []util.URLSet,
 	rtvCache *rtv.RTVCache, shouldPackage func() bool, overrideBaseURL *url.URL,
 	requireHeaders bool) (*Signer, error) {
+	var t http.RoundTripper
+	if docroot != "" {
+		t := &http.Transport{}
+		r := http.NewFileTransport(http.Dir(docroot))
```

Comment:
Clever hack. :) TIL about NewFileTransport.

If you were merging this, I'd suggest maybe passing an `http.RoundTripper` into `signer.New` instead of a docroot. That separates some of the fetching logic from signer. (In honesty, probably more of the fetching logic should be extracted out of signer, but one step at a time.) We could even imagine, like, a SQL-backed RoundTripper or something. May eventually outgrow the RoundTripper interface.

Reply:
Yes, although perhaps the transport should be a function of the URLSet Domain? (So a different client per domain, in the same way there can be a different "fetch" per domain.)

Reply:
Yeah, good point.